doc/about/api/index.md

```diff
@@ -19,11 +19,12 @@ Exceptions to the previous rule:
      `detail` or if it is private to a class type, you are not allowed to depend upon it any way. More generally,
```

Comment:
In line 1: `future-proofness`. Apparently, that's not really a word. There is `future-proof` and `future-proofing`, but not `future-proofness`.

Reply:
I came up with 

1. `# API Stability - future-proof promises of seqan3`
2. `# Future-proof API Stability`
3. `# Stability and long term promises`

Thoughts?

Reply:
I like 3, though it should be `long-term`

Reply:
> Ich finde 1. am besten. Begründung: 2. ist kaum unterscheidbar zu "# API stability" 3. weiß nicht, 1. gefällt mir besser :p

But `future-proof promises` means that the promises are future-proof, no?

Reply:
I kind of like 3 the most too. But I don't care much.

Reply:
2:1 for option three, I guess it is settled :-)